File: java-library/src/main/java/com/gblninja/tag/type/GblMetadata.java
package com.gblninja.tag.type;


import com.gblninja.tag.GblType;
import com.gblninja.tag.Tag;
import com.gblninja.tag.TagHeader;
import com.gblninja.tag.TagWithHeader;

public class GblMetadata implements Tag, TagWithHeader {
    private final TagHeader tagHeader;
    private final GblType tagType;
    private final byte[] metaData;
    private final byte[] tagData;

    public GblMetadata(TagHeader tagHeader, GblType tagType, byte[] metaData, byte[] tagData) {
        this.tagHeader = tagHeader;
        this.tagType = tagType;
        this.metaData = metaData;
        this.tagData = tagData;
    }

    @Override
    public GblType getTagType() {
        return tagType;
    }

    @Override
    public Tag copy() {
        return new GblMetadata(tagHeader, tagType, metaData, new byte[0]);
    }

    @Override
    public TagHeader getTagHeader() {
        return tagHeader;
    }

    @Override
    public byte[] getTagData() {
        return tagData;
    }

    public byte[] getMetaData() {
        return metaData;
    }
}

File: gbl/src/main/java/com/onix/gbl/parser/type/GblMetadata.kt
package parser.data.type

import parser.data.tag.TagHeader
import parser.data.tag.TagInterface

data class GblMetadata(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    val metaData: ByteArray,
    override val tagData: ByteArray
): TagInterface {
    override fun copy(): TagInterface {
        return GblMetadata(
            tagHeader = tagHeader,
            tagType = tagType,
            metaData = metaData,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }
}
